lib/src/components/NxAlert/stateful/__tests__/NxStatefulAlert.test.tsx

```diff
@@ -4,56 +4,158 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import { faCheckCircle } from '@fortawesome/free-solid-svg-icons';
+import React, { ComponentType } from 'react';
+import { faBiohazard } from '@fortawesome/free-solid-svg-icons';
 
-import * as enzymeUtils from '../../../../__testutils__/enzymeUtils';
-import 'jest-enzyme';
-import NxStatefulAlert, {
-  NxStatefulInfoAlert,
-  NxStatefulErrorAlert,
+import { rtlRender, rtlRenderElement, userEvent } from '../../../../__testutils__/rtlUtils';
+import NxStatefulAlert,
+{ NxStatefulErrorAlert,
   NxStatefulWarningAlert,
+  NxStatefulInfoAlert,
   NxStatefulSuccessAlert
 } from '../NxStatefulAlert';
 
-import NxAlert, {
-  NxInfoAlert,
-  NxErrorAlert,
-  NxWarningAlert,
-  NxSuccessAlert
-} from '../../NxAlert';
-import { JSXElementConstructor } from 'react';
+import { Props } from '../../NxAlert';
+
+describe('NxStatefulAlert', function() {
+  function baseTests<P extends Props>(
+    AlertComponent: ComponentType<P>,
+    minimalProps: P
+  ) {
+
+    // unfortunate type hacks, otherwise it wouldn't accept things like the custom className below
+    // because P could conceivably only allow a restricted set of classNames
+    const quickRender = rtlRender(AlertComponent as ComponentType<Props>, minimalProps as Props),
+        renderEl = rtlRenderElement(AlertComponent as ComponentType<Props>, minimalProps as Props);
+
+    it('renders the classNames given to it', function() {
+      const el = renderEl({ className: 'foo' }),
+          defaultEl = renderEl()!;
+
+      expect(el).toHaveClass('foo');
+
+      for (const cls of Array.from(defaultEl.classList)) {
+        expect(el).toHaveClass(cls);
+      }
+    });
+
+    it('renders the children within the top-level element', function() {
+      const children = <div data-testid="test-container"/>;
+
+      expect(quickRender({ children }).getByTestId('test-container')).toBeInTheDocument();
+    });
+
+    it('sets aria-atomic on the top-level element', function() {
+      expect(renderEl()).toHaveAttribute('aria-atomic', 'true');
+    });
+
+    it('passes any other props to the top-level element', function() {
+      const el = renderEl({ id: 'foo', title: 'bar' });
+      expect(el).toHaveAttribute('id', 'foo');
+      expect(el).toHaveAttribute('title', 'bar');
+    });
+
+    it('renders a Close button even though the onClose prop is not specified', async function() {
+      const user = userEvent.setup(),
+          view = quickRender(),
+          closeBtn = view.getByRole('button', { name: 'Close' });
+
+      expect(view.container.querySelector('.nx-alert')).toBeInTheDocument();
+      expect(closeBtn).toBeInTheDocument();
+
+      await user.click(closeBtn);
+      expect(view.container.querySelector('.nx-alert')).not.toBeInTheDocument();
```

Comment:
The `nx-alert` class is an implementation detail. IMO a better approach would be to check that the container element is empty.

Reply:
> The `nx-alert` class is an implementation detail. IMO a better approach would be to check that the container element is empty.

Changed here: c8e92ba8a07092dab3b8bb323be2bb8adb5b656b